digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/pointasset/DirectionalTrafficSignService.scala

```diff
@@ -32,14 +33,49 @@ class DirectionalTrafficSignService(val roadLinkService: RoadLinkService) extend
       asset.copy(geometry = roadLinks.find(_.linkId == asset.linkId).map(_.geometry).getOrElse(Nil))}
   }
 
+  def createFromCoordinates(incomingDirectionalTrafficSign: IncomingDirectionalTrafficSign, roadLink: RoadLink, username: String, isFloating: Boolean): Long = {
+    if(isFloating)
+      createFloatingWithoutTransaction(incomingDirectionalTrafficSign.copy(linkId = 0), username, roadLink)
+    else {
+      checkDuplicates(incomingDirectionalTrafficSign) match {
+        case Some(existingAsset) =>
+          updateWithoutTransaction(existingAsset.id, incomingDirectionalTrafficSign, roadLink, username)
+        case _ =>
+          create(incomingDirectionalTrafficSign, username, roadLink, false)
+      }
+    }
+  }
+
+  def checkDuplicates(incomingDirectionalTrafficSign: IncomingDirectionalTrafficSign): Option[DirectionalTrafficSign] = {
```

Comment:
we can create a a trait an move createFromCoordinates, checkDuplicate an all repeated method to there